### PR TITLE
chore: setup release-please

### PR DESCRIPTION
**Pull request by olavloite**
Sets up `release-please` automation for this repository.

Also fixes some license header issues with several files in the repository. Some files missed a license header, and some used a slightly different format. Now all (non-generated) files have the same license header.

### CODE REVIEW

File: .github/release-please.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+releaseType: java-yoshi
```
Comment (Vizerai) [inline .github/release-please.yml:1]: Is this for yoshi automation?

**Review by Vizerai — APPROVED** [review]
Overall LGTM.  Are you planning on setting up automated tests with yoshi similar to the client libraries?

**Comment by olavloite** [comment]
> Overall LGTM. Are you planning on setting up automated tests with yoshi similar to the client libraries?

Yes. That way we can also get it published to Maven central. See also [b/213946601](http://b/213946601)

File: .github/release-please.yml (hunk 1 above)
Comment (olavloite) [reply .github/release-please.yml:1]: Yes, this is for the Yoshi ReleasePlease bot.

**Pull request merged**